Milestone 9. Popup window details

indigodavid opened the pull request:
The current milestone adds a popup window and the following features:

- Js objects that generate the Projects and Featured Project sections from an array
- Modal window to display the highlighted project
- New links for each project

indigodavid committed:
Install linters to be executed locally

indigodavid committed:
Add projects.js with constructor methods

indigodavid committed:
Add html generators for project cards

mntedev committed:
add project modal

mntedev committed:
add project array and open modal function

indigodavid committed:
Add dynamic elements generator for projects

indigodavid committed:
Add popup functionality

In this commit is implemented the required js functions so that the
content in the Featured Project and Projects section is rendered using
Javascript. There is an array which contains the information in all
the projects. The first element of the array will appear in the
Featured project section. The rest will go to the Projects section.
If the array grows the Projects section will continue to add more cards
as needed.

Massive edit on the styling for the popup functionality. Add modal
container to accomplish this end.

indigodavid committed:
Add icons to popup links

indigodavid committed:
Refactor projects.js to comply linters

indigodavid committed:
Remove commented section on index.html

thecodechaser commented on css/style.css:
```diff
@@ -541,3 +543,75 @@ nav {
   }
 }
+
+.modal-container {
+  display: none;
+  position: fixed;
+  background-color: rgba(0, 0, 0, 0.6);
+  inset: 0 0;
+  width: 100vw;
+  height: 100vh;
+  padding: 10% 0;
+  margin: 0;
+}
```
- [ ] You did a good job on the functionalities but for this project, you are supposed to stick to the [Figma](https://www.figma.com/file/l7SqJ3ZfkAKih9sFxvWSR4/Microverse-Student-Project-1?node-id=1%3A1471) design as much as possible. Please reduce the top margin and padding from all sides for the model and make sure it matches the Figma design. Please add a blur effect in the background.
  - For blur you can use effect `backdrop-filter: blur(5px);`, use px according to your design.

### Your design

![image](https://user-images.githubusercontent.com/83803414/168031258-0e01bb3e-3293-4bf6-8f58-14f4e8ac0e01.png)

### Figma design

![image](https://user-images.githubusercontent.com/83803414/168031313-f9219b65-e89c-4ea1-9347-8aecec76e262.png)

thecodechaser replied:
- [ ] If the screen is smaller than the model, The complete model is not visible on the screen and your model is not scrollable so users can not see the complete model. Please use `overflow: scroll;` in your model controller so users can scroll the model.

thecodechaser commented on css/style.css:
```diff
@@ -541,3 +543,75 @@ nav {
+.modal-close {
+  font-size: 28px;
+  color: #67798e;
+}
```
- [ ] Please increase the close(X) button weight and make sure it matches the design.

thecodechaser commented on css/style.css:
```diff
@@ -541,3 +543,75 @@ nav {
+  margin: 0;
+}
+
+.modal {
+  background-color: #fff;
+  width: 90vw;
+  height: auto;
+  padding: 5%;
+  margin: 0 auto;
+  display: flex;
+  flex-direction: column;
+  justify-content: flex-start;
+  gap: 16px;
+}
```
- [ ] Please make sure the project description is aligned at the project's image start point and model buttons are aligned at the project's image end point and please make sure it matches with the design.

### Your design

![image](https://user-images.githubusercontent.com/83803414/168033561-c5f25b17-2626-45cc-b270-14c7823a77f7.png)

### Figma design

![image](https://user-images.githubusercontent.com/83803414/168033594-ce05649a-2dab-40ab-af13-5a2767fe9291.png)

thecodechaser commented on css/style.css:
```diff
@@ -541,3 +543,75 @@ nav {
+  cursor: pointer;
+}
+
+#live-link {
+  background-image: url(../images/see-live.svg);
+}
+
+#source-link {
+  background-image: url(../images/github.svg);
+}
```
- [ ] Please increase the font-weight for model buttons and reduce the gap between buttons and make sure it matches with Figma design.

### Your design

![image](https://user-images.githubusercontent.com/83803414/168033883-91c4326f-af7d-4624-8a1f-a875e8a390f8.png)

### Figma design 

![image](https://user-images.githubusercontent.com/83803414/168033937-80099d98-e395-4173-be54-2166ee242fcf.png)

thecodechaser reviewed (changes requested):
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @indigodavid👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- GitHub flow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️

<img src="https://media2.giphy.com/media/Jq824R93JsLwZCaiSL/giphy.gif"/>

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

indigodavid committed:
Edit styling to improve modal aspect

Add backdrop filter blur to modal container
Change flex disposition for content elements to justify-content
space-between to match the image height
Adjust the gap and font-weight between buttons in modal view
Adjust boldness of close icon
Reduce the padding
Add overflow scroll to modal container

aimemalaika reviewed (approved):
Hi @indigodavid, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

![133591850-fdcb7dd3-dd7a-4274-81e7-0e59434148f7](https://user-images.githubusercontent.com/28307960/158747852-e9f4c957-d810-4de4-b468-69b1337714ea.png)

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
